task_1/cluster_tf/cluster_autoscaler.tf

```diff
@@ -0,0 +1,223 @@
+resource "aws_iam_role_policy_attachment" "cluster_autoscaler" {
+  policy_arn = aws_iam_policy.cluster_autoscaler.arn
+  role       = module.eks.worker_iam_role_name
+}
+
+resource "aws_iam_policy" "cluster_autoscaler" {
+  name_prefix = "cluster-autoscaler"
+  description = "EKS cluster-autoscaler policy for cluster ${module.eks.cluster_id}"
+  policy      = data.aws_iam_policy_document.cluster_autoscaler.json
+}
+
+data "aws_iam_policy_document" "cluster_autoscaler" {
+  statement {
+    sid    = "clusterAutoscalerAll"
+    effect = "Allow"
+
+    actions = [
+      "autoscaling:DescribeAutoScalingGroups",
+      "autoscaling:DescribeAutoScalingInstances",
+      "autoscaling:DescribeLaunchConfigurations",
+      "autoscaling:DescribeTags",
+      "ec2:DescribeLaunchTemplateVersions",
+    ]
+
+    resources = ["*"]
+  }
+
+  statement {
+    sid    = "clusterAutoscalerOwn"
+    effect = "Allow"
+
+    actions = [
+      "autoscaling:SetDesiredCapacity",
+      "autoscaling:TerminateInstanceInAutoScalingGroup",
+      "autoscaling:UpdateAutoScalingGroup",
+    ]
+
+    resources = ["*"]
+
+    condition {
+      test     = "StringEquals"
+      variable = "autoscaling:ResourceTag/kubernetes.io/cluster/${module.eks.cluster_id}"
+      values   = ["owned"]
+    }
+
+    condition {
+      test     = "StringEquals"
+      variable = "autoscaling:ResourceTag/k8s.io/cluster-autoscaler/enabled"
+      values   = ["true"]
+    }
+  }
+}
+
+resource "kubectl_manifest" "cluster_autoscaler" {
```

Comment:
I see that we manage many Kubernetes specific resources / manifests with Terraform. What are the downsides of that approach in your opinion? Can we extract Kubernetes' related resources into a separate tool?

Reply:
> What are the downsides of that approach in your opinion?
I can think of only somewhat slowed down terraform runs but that's easy to address by splitting code in 2 or more separate repos (infrastructure provisioning, kubernetes core services provisioning).

> Can we extract Kubernetes' related resources into a separate tool?
sure we can! I would suggest to move application related resources (pods/services/configmaps etc) into separate repo managed by flux or similar, while resources rarely updated and not related to specific application could be still managed by terraform, for example: cluster autoscaler, logging/monitoring.

Reply:
Great idea, @eugenetaranov .

Actually, we do something similar in all our projects:

1. Terraform manages the infrastructure like VPC, IAM roles (as you suggested), EKS, etc.
2. We use Helmfile in a separate repo to control and configure "system-wide" addons for all EKS clusters (dev, stg, prod). That includes cluster-autoscaler, Prometheus, etc.
3. The applications are managed with a common Helm chart and a CI (separate repo).